examples/AspNetCore/TelemetryService.cs

```diff
@@ -0,0 +1,38 @@
+// <copyright file="TelemetryService.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace Examples.AspNetCore;
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+public class TelemetryService : ITelemetryService
+{
+    private const string InstrumentationScopeName = "Examples.AspNetCore";
+
+    public TelemetryService(string version)
+    {
+        this.ActivitySource = new ActivitySource(InstrumentationScopeName, version);
```

Comment:
Couple things for discussion:
1. Where do we dispose some of these properties?
2. Do we want to make sure that at maximum 1 instance can exist? (rather than assuming that folks who use DI pattern would ensure it is singleton)?

Reply:
1. A few comments regarding your first question
    1.  `TelemetryService` or `ITelemetryService` (if the is registered like `appBuilder.Services.AddSingleton<ITelemetryService>` does the interface have to implement IDisposable?) could implement `IDisposable` and I believe it will be handled on application shutdown. Is there a concern regarding order of disposal between `ActivitySource` and `TracerProvider` or `Meter` and `MeterProvider`?
    1. The examples that use statics don't dispose of `ActivitySource` and `Meter`. Is this a concern?
        1. https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/MicroserviceExample/Utils/Messaging/MessageReceiver.cs#L29
        2. https://github.com/open-telemetry/opentelemetry-dotnet/blob/c0f927f9a36ec76a76d18d7f3c1a5350dc42d316/docs/metrics/getting-started/Program.cs#L25-L26
        3. https://github.com/open-telemetry/opentelemetry-dotnet/blob/c0f927f9a36ec76a76d18d7f3c1a5350dc42d316/examples/Console/TestPrometheusExporter.cs#L27-L28

Reply:
- I'd recommend TelemetryService implement IDisposable (it is not necessary for ITelemetryService to extend IDisposable if we still have the interface at all).

- I don't think there is any concern on the relative disposal order, any order should work.

- I don't think we should try to enforce that only one instance of this object exists. For a developer unit testing their own telemetry generation it may be quite useful to have multiple instances existing even if OTel wouldn't support transmitting the telemetry sent from those multiple instances. Longer term we probably want OTel to handle situations where an app may have multiple identically named Meters existing in the process within different DI containers. These Meters might be serviced by different instances of the OTel telemetry pipeline per container, or they might be tagged in some way to allow them to be consilidated in a single pipeline.

- The reason the statics don't worry about disposal is they assume the lifetime of the Meter is the same as the lifetime of the process and the OS will release all resource on process exit. However in a DI container there is a presumption that the container might be shorter lived than the process (for example in a unit test).

Reply:
Disposal of `ActivitySource` and `Meter` has been added.